src/ports/postgres/modules/knn/knn.py_in

```diff
@@ -188,23 +190,26 @@ def kd_tree(schema_madlib, source_table, output_table, point_column_name, depth,
                                format(**locals()))
 
         output_table_tree = add_postfix(output_table, "_tree")
-        plpy.execute("CREATE TABLE {0} AS "
+        plpy.execute("CREATE TABLE {0} (tree) AS "
                      "SELECT ('{{ {1} }}')::DOUBLE PRECISION[]".
                      format(output_table_tree,
                             " ,".join(map(str, cutoffs))))
 
         case_when_clause = []
         n_leaves = pow(2, depth)
-        for i in reversed(range(n_leaves)):   # TODO: Why is this reversed?
-            cond = clauses.pop()[14:]         # TODO: Why 14?
+
+        # Traverse the clauses list in reverse order because we only want the leaf conditions
+        for i in reversed(range(n_leaves)):
+            # Remove the first 14 characters to get rid of the "WHERE 1=1 AND " segment
+            cond = clauses.pop()[14:]
```

Comment:
Can we remove the `[14:]` completely? It looks like the `WHERE` is the problem since `1=1` is a valid condition and would be `TRUE` for each row. The `WHERE` can be moved to within the query itself (line 182) instead of part of all the clauses.